### ISSUE TITLE
🐛 The number of compared files and diagnostics not shown aren't consistent

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#1201]
### Environment information

```block
main
```

### What happened?

One run

```shell
> biome format --max-diagnostics 0 lib

The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 577.
Compared 688 file(s) in 97ms
Skipped 1 file(s)
format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.


 ELIFECYCLE  Command failed with exit code 1.

```

Other run

```shell
> biome format --max-diagnostics 0 lib

The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 565.
Compared 665 file(s) in 70ms
format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.


 ELIFECYCLE  Command failed with exit code 1.
```

The numbers seem to be random. This is caused by threads accessing to the resources of `process_messages` at different times, and the resources aren't locked.

### Expected result

The numbers should always be the same at each run.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by anonrig** [comment]
I'm also seeing the same exact error.

```
❯ yarn lint:biome
yarn run v1.22.19
$ biome check .
The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 227.
Checked 2413 file(s) in 3s
Found 1 error(s)
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by Conaclos** [comment]
I think your issue is #1512
